frame/support/procedural/src/construct_runtime/expand/inherent.rs

```diff
@@ -162,12 +162,12 @@ pub fn expand_outer_inherent(
 		}
 
 		impl #scrate::traits::EnsureInherentsAreFirst<#block> for #runtime {
-			fn ensure_inherents_are_first(block: &#block) -> Result<(), u32> {
+			fn ensure_inherents_are_first(block: &#block) -> Result<Option<u32>, u32> {
 				use #scrate::inherent::ProvideInherent;
 				use #scrate::traits::{IsSubType, ExtrinsicCall};
 				use #scrate::sp_runtime::traits::Block as _;
 
-				let mut first_signed_observed = false;
+				let mut first_signed_observed = None;
```

Comment:
```suggestion
				let mut first_extrinsic_observed = None;
```
Maybe not the best name, but the term `signed` seems like a misnomer here. We return the index of the first unsigned or signed extrinsic

reasoning: If we see something unsigned we explicitly check if it `#pallet_names::is_inherent`, if its not `!is_inherent` is true and we may say this is the first index, regardless of it being signed or not. This behavior is documented in the tests

Reply:
this naming isnt better, since an inherent is also an extrinsic.

if you want to be specific here, we should name it `first-non-inherent-observed`

Reply:
Tangential hypothetical: if we where to redesign the block struct would the more appropriate name for the `extrinsics` field be `calls`? 

Inherent is an _antonym_ of Extrinsic

Reply:
Renamed to `first_non_inherent_observed` here: 9c8a11f039afdd7f3d9751abb27ae6535ac1f9da